posthog/hogql_queries/insights/trends/trends_query_runner.py

```diff
@@ -630,54 +634,57 @@ def setup_series(self) -> List[SeriesWithExtras]:
-                    new_series_data = FormulaAST(series_data).call(formula)
-
-                    new_result = group_list[0]
-                    new_result["data"] = new_series_data
-                    new_result["count"] = float(sum(new_series_data))
-                    new_result["label"] = f"Formula ({formula})"
-                    res.append(new_result)
-            return res
-
-        if len(results) > 0:
-            if self._trends_display.should_aggregate_values():
-                series_data = [[s["aggregated_value"]] for s in results]
+        def apply_formula_to_results_group(
+            results_group: List[Dict[str, Any]], aggregate_values: Optional[bool] = False
+        ) -> Dict[str, Any]:
+            """
+            Applies the formula to a list of results, resulting in a single, computed result.
+            """
+            base_result = results_group[0]
+            base_result["label"] = f"Formula ({formula})"
+            base_result["action"] = None
+
+            if aggregate_values:
+                series_data = [[s["aggregated_value"]] for s in results_group]
                 new_series_data = FormulaAST(series_data).call(formula)
-
-                new_result = results[0]
-                new_result["aggregated_value"] = float(sum(new_series_data))
-                new_result["data"] = None
-                new_result["count"] = 0
-                new_result["label"] = f"Formula ({formula})"
+                base_result["aggregated_value"] = float(sum(new_series_data))
+                base_result["data"] = None
+                base_result["count"] = 0
             else:
-                series_data = [s["data"] for s in results]
+                series_data = [s["data"] for s in results_group]
                 new_series_data = FormulaAST(series_data).call(formula)
+                base_result["data"] = new_series_data
+                base_result["count"] = float(sum(new_series_data))
 
-                new_result = results[0]
-                new_result["data"] = new_series_data
-                new_result["count"] = float(sum(new_series_data))
-                new_result["label"] = f"Formula ({formula})"
+            return base_result
 
-            return [new_result]
-        return []
+        if len(results) == 0:
+            # empty results
+            return []
+        elif self.query.trendsFilter is not None and self.query.trendsFilter.compare:
+            # with compare
+            sorted_results = sorted(results, key=itemgetter("compare_label"))
+            computed_results = []
+            for _key, group in groupby(sorted_results, key=itemgetter("compare_label")):
+                results_group = list(group)
+                aggregate_values = self._trends_display.should_aggregate_values()
+                computed_results.append(apply_formula_to_results_group(results_group, aggregate_values))
+            return computed_results
+
```

Comment:
What if there's a breakdown enabled _along with_ compare? Think we also need a test for that

Reply:
Unfortunately that doesn't work at the moment. Neither does it work for the legacy query though

Reply:
Added a fix here actually, should all work now.